lib/sles4sap_publiccloud.pm

```diff
@@ -453,4 +457,40 @@ sub cloud_file_content_replace() {
     return 1;
 }
 
+=head2 create_instance_data
+
+    Create and populate a list of publiccloud::instance and publiccloud::provider compatible
+    class instances.
+
+=cut
+
+sub create_instance_data {
+    my $provider = shift;
+    my $class = ref($provider);
+    die "Unexpected class type [$class]" unless ($class =~ /^publiccloud::(azure|ec2|gce)/);
+    my @instances = ();
+    my $inventory_file = qesap_get_inventory(get_required_var('PUBLIC_CLOUD_PROVIDER'));
+    my $ypp = YAML::PP->new;
+    my $raw_file = script_output("cat $inventory_file");
+    my $inventory_data = $ypp->load_string($raw_file)->{all}{children};
```

Comment:
Does not this block strictly depend on `qe-sap-deployment` and its Ansible inventory format?


```
my $inventory_file = qesap_get_inventory(get_required_var('PUBLIC_CLOUD_PROVIDER'));
    my $ypp = YAML::PP->new;
    my $raw_file = script_output("cat $inventory_file");
    my $inventory_data = $ypp->load_string($raw_file)->{all}{children};
```  

and maybe even more code below than this. What about to create a qesapdeployment.pm method to internally read the inventory and to return a perl data structure of some sort to be used by the double for loop below, without the need of that double loop code to know too much about the Ansible inventory format?

Reply:
I think that this code can reuse the same "data decoupling" concept, and qesapdeployment.pm method

https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/eb838e004ca8dadfec0bcd0d6baca884ada8ad3b#diff-d537e7676e4de934957a302dd9662d07237c473a02581e3fa2ca4065a6471c6dR129

Reply:
You mean some function inside qesapdeplyoment.pm that will directly 'translate' inventory.yaml into perl structure?
Yeah it can be done. I would rather do it in another PR/Jira task not to delay this one. As you can see this function was only moved elsewhere and was already existing.

Reply:
Yes it is https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/eb838e004ca8dadfec0bcd0d6baca884ada8ad3b#diff-d537e7676e4de934957a302dd9662d07237c473a02581e3fa2ca4065a6471c6dR129 code moved in a lib file. 

I'm fine to eventually look for any possible improvement in a next different PR